packages/iframe-stamper/src/index.ts

```diff
@@ -0,0 +1,192 @@
+/// <reference lib="dom" />
+
+// Header name for an API key stamp
+const stampHeaderName = "X-Stamp";
+
+type TStamp = {
+  stampHeaderName: string;
+  stampHeaderValue: string;
+};
+
+export type TIframeStamperConfig = {
+  iframeUrl: string;
+  iframeElementId: string;
+  iframeContainerId: string;
+};
+
+/**
+ * Stamper to use with `@turnkey/http`'s `TurnkeyClient`
+ * Creating a stamper inserts an iframe in the current page.
+ */
+export class IframeStamper {
+  container: HTMLElement;
+  iframe: HTMLIFrameElement;
+  iframeOrigin: string;
+  iframePublicKey: string | null;
+
+  /**
+   * Creates a new iframe stamper. This function _does not_ insert the iframe in the DOM.
+   * Call `.init()` to insert the iframe element in the DOM.
+   */
+  constructor(config: TIframeStamperConfig) {
+    if (typeof window === "undefined") {
+      throw new Error("cannot initialize iframe in non-browser environment");
+    }
+
+    if (document.getElementById(config.iframeElementId)) {
+      throw new Error(
+        `Iframe element with ID ${config.iframeElementId} already exists`
+      );
+    }
+
+    const container = document.getElementById(config.iframeContainerId);
+    if (!container) {
+      throw new Error(
+        `Cannot create iframe stamper: no container with ID ${config.iframeContainerId} exists in the current document`
+      );
+    }
+    this.container = container;
+
+    var iframe = window.document.createElement("iframe");
+    iframe.id = config.iframeElementId;
+    iframe.src = config.iframeUrl;
+    this.iframe = iframe;
+
+    const iframeUrl = new URL(config.iframeUrl);
+    this.iframeOrigin = iframeUrl.origin;
+
+    // This is populated once the iframe is ready. Call `.init()` to kick off DOM insertion!
+    this.iframePublicKey = null;
+  }
+
+  /**
+   * Inserts the iframe on the page and returns a promise resolving to the iframe's public key
+   */
+  async init(): Promise<string> {
+    this.container.appendChild(this.iframe);
+    const iframeOrigin = this.iframeOrigin;
+
+    // TODO: undo this hack.
+    const that = this;
```

Comment:
I think you could use arrow functions to get rid of this

Reply:
Yep! Good call.